File: frontend/src/types/analysis.ts
export interface EngineResult {
  engine_name: string;
  signal: 'BULLISH' | 'BEARISH' | 'NEUTRAL' | 'DETECTED' | 'CLEAR' | 'ACTIVE' | 'INACTIVE';
  confidence: number;
  details: Record<string, unknown>;
  weight: number;
}

export interface AnalysisResult {
  symbol: string;
  timeframe: string;
  timestamp: string;
  engines: EngineResult[];
  setup_quality: number;
  trade_direction: 'LONG' | 'SHORT' | 'NO_TRADE';
  ai_explanation: string;
}

export interface RiskCheckResult {
  approved: boolean;
  rejection_reason?: string;
  position_size?: number;
  risk_amount?: number;
  checks: { rule: string; passed: boolean; value: string }[];
}

export interface RiskCheckInput {
  symbol: string;
  direction: 'LONG' | 'SHORT';
  setup_quality: number;
  entry_price: number;
  stop_loss: number;
  take_profit: number;
  account_balance: number;
}

File: backend/app/schemas/analysis.py
from typing import Any, Optional

from pydantic import BaseModel

from app.schemas.market_data import OHLCVBar, NewsItemRead


class EngineOutput(BaseModel):
    engine_name: str
    signal: str
    confidence: float
    details: dict[str, Any]
    weight: float = 1.0


class AnalysisRequest(BaseModel):
    symbol: str
    timeframe: str
    bars: list[OHLCVBar]
    current_spread: Optional[float] = None
    news_items: Optional[list[NewsItemRead]] = None


class AnalysisResult(BaseModel):
    symbol: str
    timeframe: str
    setup_quality: float  # 0-100
    overall_signal: str   # LONG / SHORT / NEUTRAL
    engines: list[EngineOutput]
    aligned_signals: list[str]
    conflicting_signals: list[str]
    ai_explanation: Optional[str] = None
